py-currentscape: deploy and generate module

AurelienJaquier opened the pull request:
Hello everyone,

I have created a small module, currentscape, that allows scientists to easily produce currentscape figures, as in [this paper](https://elifesciences.org/articles/42722), and I would like to add it to the bluebrain modules so that it is easily accessible to scientists.
It my first pull request on github, so I hope i didn't do anything wrong.

Thank you,
Aurélien Jaquier

AurelienJaquier committed:
Update bbp-packages.yaml

Adding my currentscape module to bbp-packages.

AurelienJaquier committed:
Update bbp-packages.yaml

Adding my currentscape module to bbp-packages.

AurelienJaquier committed:
Update modules.yaml

Adding my currentscape package to the whitelist.

AurelienJaquier committed:
Update modules.yaml

Adding my currentscape package to the whitelist.

AurelienJaquier committed:
Create package.py

Added my currentscape package.

AurelienJaquier committed:
Create package.py

Added my currentscape package.

matz-e commented on var/spack/repos/builtin/packages/py-currentscape/package.py:
```diff
@@ -0,0 +1,23 @@
+from spack import *
+
+
+# replace all 'x-y' with 'xY' (e.g. 'Py-morph-tool' -> 'PyMorphTool')
```
```suggestion
```

matz-e commented on var/spack/repos/builtin/packages/py-currentscape/package.py:
```diff
@@ -0,0 +1,23 @@
+    git = "ssh://bbpcode.epfl.ch/cells/currentscape"
+
+    version("develop", branch="master")
+    version("0.0.1.dev0", tag="currentscape-v0.0.1.dev0", preferred=True)
```
```suggestion
    version("0.0.1.dev0", tag="currentscape-v0.0.1.dev0")
```
`preferred` is only needed when tagging an older version to be the default one.

matz-e commented on var/spack/repos/builtin/packages/py-currentscape/package.py:
```diff
@@ -0,0 +1,23 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
```
```suggestion
# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
```
Travis will complain about the wrong year.

matz-e reviewed (commented):
Overall, nice first PR!

Did you request to get added to the GH BlueBrain organization? I don't see a JIRA ticket for that, you should open one on the "Help Desk" with core services. That should then trigger our CI automatically.

Some small comments, otherwise looks nice!

AurelienJaquier committed:
Update var/spack/repos/builtin/packages/py-currentscape/package.py

Co-authored-by: Matthias Wolf <m+git@sushinara.net>

AurelienJaquier committed:
Update var/spack/repos/builtin/packages/py-currentscape/package.py

Co-authored-by: Matthias Wolf <m+git@sushinara.net>

AurelienJaquier committed:
Update var/spack/repos/builtin/packages/py-currentscape/package.py

Co-authored-by: Matthias Wolf <m+git@sushinara.net>

AurelienJaquier committed:
Update var/spack/repos/builtin/packages/py-currentscape/package.py

Co-authored-by: Matthias Wolf <m+git@sushinara.net>

AurelienJaquier committed:
Update var/spack/repos/builtin/packages/py-currentscape/package.py

Co-authored-by: Matthias Wolf <m+git@sushinara.net>

AurelienJaquier committed:
Update var/spack/repos/builtin/packages/py-currentscape/package.py

Co-authored-by: Matthias Wolf <m+git@sushinara.net>

matz-e commented:
Retest this please

Seems like Jenkins is not getting triggered.

matz-e commented on deploy/packages/bbp-packages.yaml:
```diff
@@ -128,6 +128,7 @@ packages:
       - py-bluepy@0.14.14
       - py-bluepymm@0.7.49 ^neuron%intel@19.0.4
       - py-bluepyopt@1.9.27 ^neuron%intel@19.0.4
+      - py-currentscape@0.0.1
```
It's listed as `0.0.1.dev0` below…

AurelienJaquier replied:
Hello!
I've been added to the GH BlueBrain organization, and I changed the version to 0.0.1, but the BlueBrain Deployment Test Build does not seem to launch. Is there something I'm missing?

AurelienJaquier committed:
Update package.py

Modified the version. Is it better now?

AurelienJaquier committed:
Update package.py

Modified the version. Is it better now?

pramodk commented:
retest this please

anilbey committed:
update bglibpy to 4.4.6 (#937)

AurelienJaquier commented:
Can I get some help on the BLueBrain5 Deployment Test Build failing? I don't understand the error and my search engine is not very helpful...
Also, am I allowed to relaunch a test? I've read that this FetchError can happen sometime and that just relaunching the test can solve it.

matz-e commented:
The phrase "Retest this please" should trigger another run. BTW, I looked at the project in Gerrit, and don't see any tags defined. Maybe this could be the issue for the fetch failure?

AurelienJaquier commented:
Thank you. I added a tag in my project, I hope this will solve the problem.
Retest this please

AurelienJaquier committed:
Update package.py

Changing tag to match the one from source on gerrit.

AurelienJaquier committed:
Update package.py

Changing tag to match the one from source on gerrit.

anilbey committed:
Bglibpy 4.4.10 (#938)

* update bglibpy to 4.4.6

* updated to bglpy 4.4.10

v 4.4.10 does not have the pandas, pyrsistent or the bluepy-configfile dependencies

matz-e commented:
```
==> Executing phase: 'build'
==> [2020-10-22-13:52:34.043609] '/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/external-libraries/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/python-3.7.4-tfxecymrkn/bin/python3.7' '-s' 'setup.py' '--no-user-cfg' 'build'
setup.py:3: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
import imp
Traceback (most recent call last):
File "setup.py", line 12, in <module>
with open("README.rst", encoding="utf-8") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'README.rst'
```

sergiorg-hpc committed:
Updates Libsonata package to include readers improvements (#940)

ppodhajski committed:
update glm to 0.9.9.3 (#943)

ferdonline committed:
Bump neurodamus for most recent patches (#942)

Bump neurodamus-py 2.0.0 to 2.0.2:
 - Fixing replay to work with multiple populations
 - Ensure data dir when skipping model build
 - Fix skipping synapse creation when weight is 0 (BBPBGLIB-673)
 - Fix deadlock when an exception is thrown from NEURON (BBPBGLIB-678)
 - Logging colors only for tty

Bump neurodamus-core 3.0.0 to 3.0.1:
 - Avoid getting nilSecRef from objects (HPCTM-1381)

asanin-epfl committed:
Update py-sonata-network-reduction dependencies: py-pyyaml@5.3.1, py-morph-tool@0.2.10 (#930)

AurelienJaquier committed:
Update bbp-packages.yaml

Updated currentscape version.

AurelienJaquier committed:
Update bbp-packages.yaml

Updated currentscape version.

AurelienJaquier committed:
Update package.py

Updated version & tag.

AurelienJaquier committed:
Update package.py

Updated version & tag.

AurelienJaquier commented:
Thank you Matthias for your help. I modified my files on bbpcode to avoid the FileNotFoundError. But it is failing again for a reason I don't understand ^^' . Maybe it will work with a simple Retest this please ?

NadirRoGue committed:
Brion and Brayns are dependent on GLM (#944)

Commit:
Steps updates (#941)

* gmsh: add version 4.6.0
* omega-h: new version 9.32.5.dev3
* steps: new test requirements

sergiorg-hpc committed:
libsonata-report: Improves initialization performance (#945)

ppodhajski committed:
adapt brion test to a new python module name (#946)

pramodk committed:
Adding nvidia-hpc-sdk based on upstream PR (#935)

* New compiler: nvhpc (NVIDIA HPC SDK) (https://github.com/spack/spack/pull/19294)
* Add nvhpc compiler definition: "spack compiler add" will now look
  for instances of the NVIDIA HPC SDK compiler executables
  (nvc, nvc++, nvfortran) in supplied paths
* Add the nvhpc package which installs the nvhpc compiler
* Add testing for nvhpc detection and C++-standard/pic flags

Based on https://github.com/spack/spack/pull/19294

* Add CUDA@11 required for latest NVIDIA-HPC-SDK
* Fix legacy apis : setup_environment to setup_run_environment

pramodk committed:
NEURON and CoreNEURON should use legacy units for BBP/HBP deployment (#947)

matz-e commented:
Retest this please

I've cleaned out the PR directory, let's give this another try

matz-e commented:
OK, finally noticed what's going on. The module generation fails, as upstream has added dependencies that are not present on your branch. If you rebase on `develop`, this should be successful.

Commit:
Merge branch 'develop' of https://github.com/AurelienJaquier/spack into develop

matz-e reviewed (approved):
Hi, `git rebase bbp/develop` (or whatever you call our upstream) would have been cleaner to look at, but I think the squash-and-merge should take care of this.

In the future, you should be able to create branches in this repository, too. Then others can help out if quick fixes are needed.

AurelienJaquier commented:
It worked! Thank you very much for your help!
